examples/next-login/src/pages/home/signup.tsx

```diff
@@ -0,0 +1,157 @@
+    const computedAge = Math.floor(timeDiff / (1000 * 3600 * 24) / 365.25)
+    setAge(computedAge)
+    if (computedAge < 1) {
+      setTooYoung(true)
+      setAlertSubmit('error')
+    } else {
+      setTooYoung(false)
+      setValues(val)
+      setAlertSubmit('success')
+      setTimeout(() => {
+        setAlertSubmit('')
+      }, 5000)
+
+      console.log('Too young ? ', tooYoung)
+      console.log('Values', values)
+    }
+  }
+  const displayAlert = () => {
+    if (alertSubmit === 'success') {
+      return (
+        <Alert sentiment="success" title="Account Created" closable>
+          Welcome {values.gender} {values.firstname} {values.lastname}, {age}{' '}
+          year{age > 1 ? 's' : ''} old. Your email adress is {values.email}.{' '}
+        </Alert>
+      )
+    }
+    if (alertSubmit === 'error') {
+      return (
+        <Alert sentiment="danger" closable>
+          Must be 1+
+        </Alert>
+      )
+    }
+    return <></>
+  }
+  return (
+    <StyledSigninContainer>
+      <Form<FormValues>
+        methods={methods}
+        errors={mockErrors}
+        onRawSubmit={handleSubmit}
+      >
+        <Stack gap={1} alignItems="center">
+          <Icon name="profile" size="1.7em" />
+          <Text as="div" placement="center" variant="heading">
+            Sign up form
+          </Text>
+          <Stack width="100%">
+            <RadioGroupField name="gender" direction="row">
+              <RadioGroupField.Radio name="mr" value="mr" label="Mr" />
+              <RadioGroupField.Radio name="mrs" value="mrs" label="Mrs" />
+            </RadioGroupField>
+            <Stack gap={3} direction="row">
+              <StyledInput
+                required
+                label="First Name"
+                name="firstname"
+                placeholder="John"
+                className="inputs"
+              />
+              <StyledInput
+                required
+                label="Last Name"
+                name="lastname"
+                placeholder="Smith"
+                className="inputs"
+              />
+            </Stack>
+            <DateField
+              name="birthdate"
+              label="Birth Date"
+              required
+              className="age-input"
+            />
+            <StyledInput
+              label="Email"
+              name="email"
+              required
+              placeholder="example@email.com"
+              rules={{
+                pattern: { value: EMAIL_REGEX, message: 'Invalid format' },
+              }}
+              className="inputs"
+            />
+            <StyledInput
+              label="Password"
+              name="password"
+              type="password"
+              required
+              minLength={8}
+              helper="Min 8 characters"
+            />
+          </Stack>
+          <Submit>Create an account</Submit>
+        </Stack>
+      </Form>
+      {displayAlert()}
```

Comment:
Looking at this function you could probably convert it to a component :) like <DisplayAlert/> and use some props to passe the state down to it.
Just a suggestion for readability :)